pandas/core/arrays/categorical.py

```diff
@@ -2007,9 +2007,10 @@ def __getitem__(self, key):
         if com.is_bool_indexer(key):
             key = check_bool_array_indexer(self, key)
 
-        return self._constructor(
-            values=self._codes[key], dtype=self.dtype, fastpath=True
-        )
+        result = self._codes[key]
+        if result.ndim > 1:
+            return result
```

Comment:
This should also raise a warning? 

(but adding the warning is not a blocker for the RC, since that's a future deprecation, as long as the behaviour change is already there)

Reply:
Actually, we are returning plain integer codes here, which doesn't look good.

And checked with 0.25, and there multi-dim indexing on categorical/categoricalindex basically fails (it "returns", but whenever you do something it gives an error; even converting to the result to a numpy array (the original matplotlib use case for the 2d indexing) fails). 
So I would also raise an error here, like you did for some other arrays.

Reply:
Agreed this should raise to match Interval. Will push an update.

Reply:
It's not straightforward, as actually this already gets catched before in the boolean checking ..

Reply:
(I would be fine with doing a clean-up for my comments in a follow-up PR)

Reply:
Yeah, this is a bit tricky. Will look a bit longer.

Reply:
this raised an error in 0.25, it raises a different (but wrong) error now, so I think fine to do in a follow-up

Reply:
I'm very confused by what's going on with with `Series.__getitem__` here. Are we OK with changing this for categorical after the RC?